cpp/benchmarks/type_dispatcher/type_dispatcher_benchmark.cu

```diff
@@ -0,0 +1,231 @@
+// n_rows * n_cols. 
+template<FunctorType functor_type>
+struct RowHandle {
+  template<typename T>
+  __device__ void operator()(const gdf_column& source, gdf_index_type index){
+    using F = Functor<T, functor_type>;
+    static_cast<T*>(source.data)[index] = 
+      F::f(static_cast<T*>(source.data)[index]);
+  }
+};
+
+// This is for DEVICE_DISPATCHING
+template<FunctorType functor_type>
+__global__ void device_dispatching_kernel(device_table source){
+
+  const gdf_index_type n_rows = source.num_rows();
+  gdf_index_type index = threadIdx.x + blockIdx.x * blockDim.x;
+  
+  while(index < n_rows){
+    for(gdf_size_type i = 0; i < source.num_columns(); i++){
+      cudf::type_dispatcher(source.get_column(i)->dtype,
+                          RowHandle<functor_type>{}, *source.get_column(i), index);
+    }
+    index += blockDim.x * gridDim.x;
+  } // while
+}
+
+template<FunctorType functor_type, DispatchingType dispatching_type, class T>
+void launch_kernel(cudf::table& input, T** d_ptr, int work_per_thread){
+  
+  const gdf_size_type n_rows = input.num_rows();
+  const gdf_size_type n_cols = input.num_columns();
+    
+  int grid_size = launch_configuration(n_rows, work_per_thread);
+ 
+  if(dispatching_type == HOST_DISPATCHING){
+    // std::vector<cudf::util::cuda::scoped_stream> v_stream(n_cols);
+    for(int c = 0; c < n_cols; c++){
+      cudf::type_dispatcher(input.get_column(c)->dtype, ColumnHandle<functor_type>{}, input.get_column(c), work_per_thread);
+    }
+  }else if(dispatching_type == DEVICE_DISPATCHING){
+    auto d_source_table = device_table::create(input);
+    auto f = device_dispatching_kernel<functor_type>;
+    // Launch the kernel
+    f<<<grid_size, block_size>>>(*d_source_table);
+  }else if(dispatching_type == NO_DISPATCHING){
+    auto f = no_dispatching_kernel<functor_type, T>;
+    // Launch the kernel
+    f<<<grid_size, block_size>>>(d_ptr, n_rows, n_cols);
+  }
+}
+
+template<class TypeParam, FunctorType functor_type, DispatchingType dispatching_type>
+void type_dispatcher_benchmark(benchmark::State& state){
+  const gdf_size_type source_size = static_cast<gdf_size_type>(state.range(1));
+  
+  const gdf_size_type n_cols = static_cast<gdf_size_type>(state.range(0));
+  
+  const gdf_size_type work_per_thread = static_cast<gdf_size_type>(state.range(2));
+
+  
+  std::vector<cudf::test::column_wrapper<TypeParam>> v_src(
+      n_cols,
+      {
+        source_size,
+        [](gdf_index_type row){ return static_cast<TypeParam>(row); },
+        [](gdf_index_type row) { return true; }
+      }
+  );
+  
+  std::vector<gdf_column*> vp_src(n_cols);
+  for(size_t i = 0; i < v_src.size(); i++){
+    vp_src[i] = v_src[i].get();  
+  }
+
+  cudf::table source_table{ vp_src };
+  
+  // For no dispatching:
+  std::vector<TypeParam*> h_vec(n_cols);
+  TypeParam** d_ptr = nullptr;
+  
+  if(dispatching_type == NO_DISPATCHING){
+    RMM_TRY(RMM_ALLOC(&d_ptr, sizeof(TypeParam*)*n_cols, 0));
```

Comment:
How about using a `rmm::device_vector<TypeParam*>` which points to the contents of `rmm::device_vector<TypeParam>` instead to avoid all the manual memory allocations?